### PR TITLE
Adding compatibility to Swift 1.2

### LINKED ISSUE #151: Swift 1.2 compile issues
Lots of compile issues with Swift 1.2 (not unique to this library, FWIW).  Tried the auto-convert to 1.2 and that did not resolve them all.

Tried fixing other manually but not familiar enough with the 1.2 migration issues to work through it.

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #151

### CODE REVIEW

**Comment by dautermann** [comment]
THANKS for making this pull request available.  I just started using Xcode 6.3 and promptly ran into issues with HenekeSwift.  

The only comment I'd want to make is that I wonder if there should be compiler conditionals (e.g. "`#ifdef` something) in the code you've added to continue using the pre-Swift-1.2 calls/API's for anyone not yet using Xcode 6.3?

**Comment by marcelofabri** [comment]
I don't think this is worth the effort. IMO, we should have an `xcode-6.3` (or something like it) branch and only merge it on `master` when Xcode 6.3 is released.

If someone needs to keep using an older version of Xcode, just stick with an older version of HanekeSwift.

File: Haneke/CryptoSwiftMD5.swift [hunk 1]
```diff
@@ -89,55 +94,55 @@ func rotateLeft(v:UInt32, n:UInt32) -> UInt32 {
...
+        
         // hash values
         var hh = h
         
         // Step 2. Append Length a 64-bit representation of lengthInBits
         var lengthInBits = (message.length * 8)
         var lengthBytes = lengthInBits.bytes(64 / 8)
-        tmpMessage.appendBytes(reverse(lengthBytes))
-
+        tmpMessage.appendBytes(reverse(lengthBytes));
```
Comment (dautermann) [inline Haneke/CryptoSwiftMD5.swift]: any reason for the "`;`" ?

Comment (marcelofabri) [inline Haneke/CryptoSwiftMD5.swift]: I imported this again from [CryptoSwift](https://github.com/krzyzanowskim/CryptoSwift). 

But I would say that there's no reason.

**Comment by arnaudmeunier** [comment]
Thanks for the PR! You might to consider adding a podspec for CocoaPods (cf podspec branch - https://github.com/Haneke/HanekeSwift/blob/podspec/HanekeSwift.podspec)

File: Haneke/Cache.swift [hunk 2]
```diff
@@ -78,12 +80,11 @@ public class Cache<T : DataConvertible where T.Result == T, T : DataRepresentabl
     public func fetch(#key : String, formatName : String = HanekeGlobals.Cache.OriginalFormatName, failure fail : Fetch<T>.Failer? = nil, success succeed : Fetch<T>.Succeeder? = nil) -> Fetch<T> {
         let fetch = Cache.buildFetch(failure: fail, success: succeed)
         if let (format, memoryCache, diskCache) = self.formats[formatName] {
-            if let wrapper = memoryCache.objectForKey(key) as? ObjectWrapper {
-                if let result = wrapper.value as? T {
+            if let wrapper = memoryCache.objectForKey(key) as? ObjectWrapper,
+                let result = wrapper.value as? T {
                     fetch.succeed(result)
-                    diskCache.updateAccessDate(dataFromValue(result, format: format), key: key)
+                    diskCache.updateAccessDate(self.dataFromValue(result, format: format), key: key)
```
Comment (callmewhy) [inline Haneke/Cache.swift]: why need self here?

File: Haneke/DiskCache.swift [hunk 3]
```diff
@@ -20,9 +20,9 @@ public class DiskCache {
     
     public let path : String
 
-    public var size : UInt64 = 0
+    public var size : Int = 0
```
Comment (callmewhy) [inline Haneke/DiskCache.swift]: Why use Int instead of UInt64?

**Comment by barnybug** [comment]
+1

**Comment by mkoppanen** [comment]
XCode 6.3 has now been released so this is more relevant.

**Comment by hpique** [comment]
@marcelofabri Thanks for making this PR available while I was away. Reviewing it now.

File: Haneke/DiskCache.swift (hunk 3 above)
Comment (hpique) [inline Haneke/DiskCache.swift]: Wondering the same. 

@marcelofabri Changing this back to UInt64.

**Comment by marcelofabri** [comment]
:+1:

**Comment by mkoppanen** [comment]
:+1:

**Comment by vchoubard** [comment]
:+1:

**Pull request merged**